festim/stepsize.py

```diff
@@ -96,6 +96,9 @@ def adapt(self, t, nb_it, converged):
             max_stepsize = self.adaptive_stepsize["max_stepsize"]
 
             if not converged:
+                if dt_min is None:
+                    raise ValueError("Solver diverged but dt_min is not set.")
```

Comment:
The fix looks ok to me. Maybe, combine if-clauses? `dt_min` has to exists for the second check.

Reply:
yes, I understand.
I meant smth like that so the checks would be placed together:
```python
if dt_min is None:
    ...
elif float(self.value) < dt_min:
    ...
```